src/codegen.cpp

```diff
@@ -2903,8 +2903,11 @@ static bool emit_builtin_call(jl_codectx_t &ctx, jl_cgval_t *ret, jl_value_t *f,
                                 emit_datatype_nfields(ctx, emit_typeof_boxed(ctx, obj)),
                                 jl_true);
                         }
+                        bool isboxed = !jl_datatype_isinlinealloc(jt);
                         Value *ptr = maybe_decay_tracked(data_pointer(ctx, obj));
-                        *ret = typed_load(ctx, ptr, vidx, jt, obj.tbaa, nullptr, false);
+                        *ret = typed_load(ctx, ptr, vidx,
```

Comment:
The return cgval gets annotated with the type we pass in. Does widening to `Any` pessimize later codegen?

Reply:
I'd guess not really, since codegen generally uses inferred types when it needs type info. We're also doing the same thing for arrayref on pointer arrays, so it'd better work :stuck_out_tongue: